tests-integration/src/runvm.rs

```diff
@@ -0,0 +1,165 @@
+use anyhow::{Context, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use clap::Subcommand;
+use fn_error_context::context;
+use xshell::{cmd, Shell};
+
+const BUILDER_ANNOTATION: &str = "bootc.diskimage-builder";
+const TEST_IMAGE: &str = "localhost/bootc";
+const TESTVMDIR: &str = "testvm";
+const DISK_CACHE: &str = "disk.qcow2";
+const IMAGEID_XATTR: &str = "user.bootc.container-image-digest";
+
+#[derive(Debug, Subcommand)]
+#[clap(rename_all = "kebab-case")]
+pub(crate) enum Opt {
+    PrepareTmt {
+        #[clap(long)]
+        /// The container image to spawn, otherwise one will be built
+        testimage: Option<String>,
+    },
+    CreateQcow2 {
+        /// Input container image
+        container: String,
+        /// Write disk to this path
+        disk: Utf8PathBuf,
+    },
+}
+
+struct TestContext {
+    sh: xshell::Shell,
+    targetdir: Utf8PathBuf,
+}
+
+fn image_digest(sh: &Shell, cimage: &str) -> Result<String> {
+    let key = "{{ .Digest }}";
+    let r = cmd!(sh, "podman inspect --type image --format {key} {cimage}").read()?;
+    Ok(r)
+}
+
+fn builder_from_image(sh: &Shell, cimage: &str) -> Result<String> {
+    let mut inspect: serde_json::Value =
+        serde_json::from_str(&cmd!(sh, "podman inspect --type image {cimage}").read()?)?;
+    let inspect = inspect
+        .as_array_mut()
+        .and_then(|v| v.pop())
+        .ok_or_else(|| anyhow::anyhow!("Failed to parse inspect output"))?;
+    let config = inspect
+        .get("Config")
+        .ok_or_else(|| anyhow::anyhow!("Missing config"))?;
+    let config: oci_spec::image::Config =
+        serde_json::from_value(config.clone()).context("Parsing config")?;
+    let builder = config
+        .labels()
+        .as_ref()
+        .and_then(|l| l.get(BUILDER_ANNOTATION))
+        .ok_or_else(|| anyhow::anyhow!("Missing {BUILDER_ANNOTATION}"))?;
+    Ok(builder.to_owned())
+}
+
+#[context("Running bootc-image-builder")]
```

Comment:
Can we use `bootc install to-disk` to generate disk image?

Reply:
Yeah, but I think down the line we'll migrate bib to be more of a SDK (cc https://gitlab.com/fedora/bootc/tracker/-/issues/2 ) and so we might as well start pulling a container here.